### ISSUE TITLE
About the mask in the BPR loss function

### ISSUE THREAD

**Issue by GXNU156489** [tsinghua-fib-lab/DICE#7]
Hello, author. I am a novice who has just entered this field. I found that there are 2 BRP loss functions in your code, and one of the loss functions has a parameter of mask. I would like to ask you about the function of the mask parameter and the generation conditions. Thank you and look forward to your reply.
![image](https://github.com/tsinghua-fib-lab/DICE/assets/36919834/625bdc73-d3f0-4683-8c21-19452e5e0311)

**Comment by DavyMorgan** [comment]
Hi, the mask_bpr_loss is adopted in our DICE model
```
loss_int = self.mask_bpr_loss(p_score_int, n_score_int, mask)
loss_pop = self.mask_bpr_loss(n_score_pop, p_score_pop, mask) + self.mask_bpr_loss(p_score_pop, n_score_pop, ~mask)
```

the mask here indicates the training sample pairs where the negative item has a larger popularity value than the positive item, which fits the collider effect in causal inference. In other words, for training samples with `mask=True`, the user-item interaction is caused by interest of high probability, since the conformity towards the negative item is actually stronger than the positive item.

**Comment by GXNU156489** [comment]
> 嗨，我们的DICE模型采用了mask_bpr_loss
> 
> ```
> loss_int = self.mask_bpr_loss(p_score_int, n_score_int, mask)
> loss_pop = self.mask_bpr_loss(n_score_pop, p_score_pop, mask) + self.mask_bpr_loss(p_score_pop, n_score_pop, ~mask)
> ```
> 
> 这里的 mask 表示负项比正项具有更大流行度值的训练样本对，符合因果推理中的碰撞效应。换句话说，对于具有 的训练样本`mask=True`，用户-项目交互是由高概率兴趣引起的，因为对负项目的一致性实际上强于正项目。

hi，Thanks for your answer, I've figured it out
